File: frontend/src/app/dashboard/model/dto/dashboard-creation.dto.ts
import { ChartCreationOptionsDto } from './chart-creation-options.dto';
import { CompanyId, DashboardId, UserId } from '../../../shared/model/id-types';

/**
 * Used to deliver data from Frontend to Backend.
 * To transport data from Backend to Frontend, see the DashboardDto.
 * DashboardId and date of creation are not set in the Frontend but in the Backend (for new Dashboards).
 */
export interface DashboardCreationDto {
  id?: DashboardId;
  title: string;
  creatorId?: UserId;
  companyId: CompanyId;
  chartCreationOptions: ChartCreationOptionsDto[];
}

File: backend/burning-okr/burning-okr-app/src/main/java/org/burningokr/dto/dashboard/creation/DashboardCreationDto.java
package org.burningokr.dto.dashboard.creation;

import java.util.Collection;
import java.util.UUID;
import lombok.Data;

@Data
public class DashboardCreationDto {
  private Long id;
  private String title;
  private UUID creatorId;
  private Long companyId;
  private Collection<ChartCreationOptionsDto> chartCreationOptions;
}
